src/frontend/src/optimizer/plan_node/logical_over_agg.rs

```diff
@@ -243,13 +157,23 @@ impl LogicalOverAgg {
 
                 // override the frame
                 // TODO(rc): We can only do the optimization for constant offset.
-                Some(if window_function.kind == WindowFuncKind::Lag {
+                if window_function.kind == WindowFuncKind::Lag {
                     Frame::Rows(FrameBound::Preceding(offset), FrameBound::CurrentRow)
                 } else {
                     Frame::Rows(FrameBound::CurrentRow, FrameBound::Following(offset))
-                })
+                }
             }
-            _ => window_function.frame,
+            _ => window_function.frame.unwrap_or({
```

Comment:
Can we make it more explicit? 👀

```suggestion
            WindowFuncKind::Agg => window_function.frame.unwrap_or({
```

Reply:
OK